Lib/_pydatetime.py

```diff
@@ -223,7 +223,7 @@ def _can_support_c99():
             _supports_c99 = (
                 _time.strftime("%F", (1900, 1, 1, 0, 0, 0, 0, 1, 0)) == "1900-01-01")
         except ValueError:
-            pass
+            return False
```

Comment:
I think it should not only return False, but cache it in `_supports_c99`. Otherwise this runtime check will be performed again and again. If it failed the first time, no reasons to expect that it will fail next time.

Reply:
Not sure how I missed it. Thanks!